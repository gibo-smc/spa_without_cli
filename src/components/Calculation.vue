<template lang="pug">
div
  p
    input.left(type="text" v-model.number="leftInput")
    span
      | +
    input.right(type="text" v-model.number="rightInput")
    span
      | =
    span.answer
      | {{ answer }}
  button.exec(@click="execCalc")
    | 足し算実行
</template>
<script>
import addition from '../modules/addition'
export default {
  name: 'Calculation',
  data () {
    return {
      leftInput: 0,
      rightInput: 0,
      answer: 0
    }
  },
  methods: {
    execCalc () {
      this.answer = addition(this.leftInput, this.rightInput)
    }
  }
}
</script>

<style lang="sass" scoped>
p
  display: flex
  input
    width: 30px
    margin-right: 14px
  span
    margin-right: 14px
</style>
