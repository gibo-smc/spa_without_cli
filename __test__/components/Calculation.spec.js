import { mount } from '@vue/test-utils'
import Calculation from '@/components/Calculation'

describe('Calculation', () => {
  const wrapper = mount(Calculation)
  it('計算ボタンが存在すること', () => {
    expect(wrapper.contains('button.exec')).toBeTruthy()
  })
})
