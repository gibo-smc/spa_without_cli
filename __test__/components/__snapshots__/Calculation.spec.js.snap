// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculation 数字を入力し計算ボタンを押下すると正しく計算されること 1`] = `
<div>
  <p><input type="text" class="left"><span>+</span><input type="text" class="right"><span>=</span><span class="answer">7</span></p><button class="exec">足し算実行</button>
</div>
`;
